tools/models/model_factory.py

```diff
@@ -0,0 +1,168 @@
+import os, sys
+import torch
+import torch.optim as optim
+from inSituML.utilities import MMD_multiscale, fit, load_checkpoint
+from inSituML.ks_models import INNModel
+
+from inSituML.args_transform import MAPPING_TO_LOSS
+from inSituML.encoder_decoder import Encoder
+from inSituML.encoder_decoder import Conv3DDecoder
+from inSituML.loss_functions import EarthMoversLoss
+from inSituML.networks import VAE
+from models.architectures import ModelFinal
+
+def get_world_size():
```

Comment:
This is unrelated to model definitions. The main program should still handle obtaining this value. The function can be moved into a utility module, but it should be a module with parallel-training utilities.